src/unix/linux_like/android/b32/x86/mod.rs

```diff
@@ -574,6 +574,40 @@ pub const REG_EFL: ::c_int = 16;
 pub const REG_UESP: ::c_int = 17;
 pub const REG_SS: ::c_int = 18;
 
+// socketcall values from linux/net.h (only the needed ones, and not public)
+const SYS_ACCEPT4: ::c_int = 18;
+
+f! {
+    // Sadly, Android before 5.0 (API level 21), the accept4 syscall is not
+    // exposed by the libc. As work-around, we implement it as raw syscall.
+    // Note that for x86, the `accept4` syscall is not available either,
+    // and we must use the `socketcall` syscall instead.
+    // This workaround can be removed if the minimum Android version is bumped.
+    // When the workaround is removed, `accept4` can be moved back
+    // to `linux_like/mod.rs`
```

Comment:
From my understanding, the  `socketcall` syscall needs to be made on x86 Android even on higher API versions, so that comment doesn't seem to be correct.

Further, a reference to the seccomp profile at https://android.googlesource.com/platform/bionic.git/+/master/libc/SYSCALLS.TXT might also be helpful?

Reply:
For higher API levels, libc already exposes `accept4`. I would expect the system libc to implement that correctly.

Reply:
To clarify: rather than a function in Rust `libc`, it would just be an extern fn calling into the system libc, which should work out of the box.

Reply:
Maybe I'm a bit confused, but isn't that what I tried on https://github.com/rust-lang/rust/issues/82400#issuecomment-783405484, using `libc::accept4` directly on an Android 6.0 x86 emulator?

Reply:
I don't think so, because that was calling the function that this PR replaces for x86.

If you want to call `accept4` directly from the system libc, you have to declare it in an `extern "C"` block and call that :)

Reply:
I can see clearly now the rain is gone :-)